policy-engine/src/config/settings.rs

```diff
@@ -73,19 +73,20 @@ impl AppSettings {
     }
 
     /// Validate and return policy plugins.
-    pub fn policy_plugins(&self) -> Fallible<Vec<BoxedPlugin>> {
-        let mut plugins = Vec::with_capacity(self.policies.len());
-        for conf in &self.policies {
-            let plugin = conf.build_plugin()?;
-            plugins.push(plugin);
-        }
-
-        // TODO(lucab): drop this as soon as all config-maps are in place (prod & staging).
-        if plugins.is_empty() {
-            plugins = default_openshift_plugins();
-        }
+    pub fn policy_plugins(
+        &self,
+        registry: Option<&prometheus::Registry>,
+    ) -> Fallible<Vec<BoxedPlugin>> {
+        // TODO(steveeJ):  prevent this call in case it's not required later
```

Comment:
In the spirit of this TODO, perhaps in the short term it is better to move the `?` in the non-empty conditional branch below?

Reply:
Unpacking the result in the if scope creates the same issue as moving the call there altogether:

```
error[E0716]: temporary value dropped while borrowed
  --> policy-engine/src/config/settings.rs:83:14
   |
82 |         let policies: &Vec<Box<dyn PluginSettings>> = if self.policies.is_empty() {
   |             -------- borrow later stored here
83 |             &default_policies?
   |              ^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
84 |         } else {
   |         - temporary value is freed at the end of this statement
```

Reply:
Ack, let's keep it as is for the moment.